components/dashboard/src/workspaces/StartWorkspaceModal.tsx

```diff
@@ -4,70 +4,21 @@
-
-    const list = (selection === 'Recent' ? p.recent : p.examples).map((e, i) =>
-        <a key={`item-${i}-${e.title}`} href={e.startUrl} className="rounded-xl group hover:bg-gray-100 dark:hover:bg-gray-800 flex p-4 my-1">
-            <div className="w-full">
-                <p className="text-base text-gray-800 dark:text-gray-200 font-semibold">{e.title}</p>
-                <p>{e.description}</p>
-            </div>
-        </a>);
-
-    return <Modal onClose={p.onClose} visible={p.visible}>
+export function StartWorkspaceModal(props: StartWorkspaceModalProps) {
+    return <Modal onClose={props.onClose} onEnter={() => false} visible={props.visible}>
         <h3 className="pb-2">New Workspace</h3>
         {/* separator */}
-        <div className="border-t border-gray-200 dark:border-gray-800 mt-2 -mx-6 px-6 pt-2">
-            <div className="flex">
-                <TabMenuItem name='Recent' selected={selection === 'Recent'} onClick={() => setSelection('Recent')} />
-                {p.examples.length>0 && <TabMenuItem name='Examples' selected={selection === 'Examples'} onClick={() => setSelection('Examples')} />}
-            </div>
-        </div>
-        <div className="border-t border-gray-200 dark:border-gray-800 -mx-6 px-6 py-2">
-            {list.length > 0 ?
-                <p className="my-4 text-base">
-                    {selection === 'Recent' ?
-                        'Create a new workspace using the default branch.' :
-                        'Create a new workspace using an example project.'}
-                </p> : <p className="h-6 my-4"></p>}
-            <div className="space-y-2 mt-4 overflow-y-scroll h-80 pr-2">
-                {list.length > 0 ? list :
-                    (selection === 'Recent' ?
-                        <div className="flex flex-col pt-10 items-center px-2">
-                            <h3 className="mb-2 text-gray-500 dark:text-gray-400">No Recent Projects</h3>
-                            <p className="text-center">Projects you use frequently will show up here.</p>
-                            <p className="text-center">Prefix a Git repository URL with {window.location.host}/# or start with an example.</p>
-                            <button onClick={() => setSelection('Examples')} className="font-medium mt-8">Select Example</button>
-                        </div> :
-                        <div className="flex flex-col pt-10 items-center px-2">
-                            <h3 className="mb-2 text-gray-500 dark:text-gray-400">No Example Projects</h3>
-                            <p className="text-center">Sorry there seem to be no example projects, that work with your current Git provider.</p>
-                        </div>)
-                }
-            </div>
+        <div className="border-t border-gray-200 dark:border-gray-800 mt-2 -mx-6 px-6 pt-4">
+            <h4 className="text-base">Search or paste a repository URL</h4>
+            <RepositoryFinder />
```

Comment:
**question**: Could it help to keep the input state here when closing and re-opening the modal? Sometimes you could accidentally close the modal just before clicking the result.

Reply:
I would vote against this, as I prefer the modal to always have a ready-to-go fresh start as opposed to "old stuff you need to clean up first".

Can it really happen that users close the modal by mistake and then try to open it again? (Like, accidentally press `Esc`, or accidentally click the `X`? 🤔 I find this scenario a bit unlikely.)

Reply:
I've seen both approaches work equally good depending on the application.

> Can it really happen that users close the modal by mistake and then try to open it again?

I'd think yes, but this will become much more relevant once the command palette (this modal) becomes more useful or contains more actions to take. However, let's leave this as is for now and come back to this if we run into this in action. 🤝